bee-identity/src/identity.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Holds the `Identity` type.
+
+use std::{fmt, path::Path};
+
+use libp2p_core::{identity::PublicKey, PeerId};
+
+use crate::{migration, pem_file, Keypair as Ed25519Keypair, PublicKey as Ed25519PublicKey};
+
+/// Represents a node identity.
+#[derive(Clone, Debug)]
+pub struct Identity {
+    keypair: Ed25519Keypair,
+    restored_from_file: bool,
+}
+
+impl Identity {
+    /// Creates a new identity.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Restores an [`Identity`] from a [`Keypair`](crate::ed25519::Keypair).
+    pub fn from_keypair(keypair: Ed25519Keypair) -> Self {
+        Self {
+            keypair,
+            restored_from_file: true,
```

Comment:
True ?

Reply:
It's a bit weird to me that this is called `restored_from_file` when obviously it can be restored from pretty much everything

Reply:
you're right. `restored` is enough.

Reply:
Changed.